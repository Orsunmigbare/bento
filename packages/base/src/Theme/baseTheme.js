import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
