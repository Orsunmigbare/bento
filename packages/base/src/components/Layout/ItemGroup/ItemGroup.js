import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { spacing as space, themify } from "@staccx/theme"

const ItemGroup = ({ children, spacing }) => (
  <Wrap>
    <ItemGroupElement spacing={spacing}>{children}</ItemGroupElement>
  </Wrap>
)
export const ITEM_GROUP_WRAPPER = "ITEM_GROUP_WRAPPER"
const Wrap = styled.div`
  overflow: hidden;
  ${themify(ITEM_GROUP_WRAPPER)};
`

export const ITEM_GROUP = "ITEM_GROUP"
const ItemGroupElement = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 -${p => space(p.spacing)} -${p => p.spacing};
  & > * {
    margin: 0 ${p => space(p.spacing)} ${p => p.spacing};
  }
  ${themify(ITEM_GROUP)};
`

ItemGroup.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
    PropTypes.array
  ]).isRequired,
  spacing: PropTypes.oneOf([
    "micro",
    "tiny",
    "small",
    "medium",
    "mediumPlus",
    "large",
    "huge"
  ])
}

ItemGroup.defaultProps = {
  spacing: "tiny"
}

export default ItemGroup
