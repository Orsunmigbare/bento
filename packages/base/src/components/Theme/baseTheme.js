export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
