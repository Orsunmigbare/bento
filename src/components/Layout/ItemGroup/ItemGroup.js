import React from "react"
import styled from "styled-components"
import { applyVariants, spacing, commonPropTypes } from "../../../theming"
import themeProps from "./ItemGroup.themeProps"
import { componentCreateFactory } from "../../../theming/utils/createVariantsFunctionFactory"
/**
 * The ItemGroup object creates spacing between all its children and wraps the items when there is no room
 */
const ItemGroup = ({ children, spacing, variant, className, ...restProps }) => (
  <Wrap variant={variant} className={className} {...restProps}>
    <ItemGroupElement spacing={spacing} variant={variant}>
      {children}
    </ItemGroupElement>
  </Wrap>
)

const Wrap = styled.div`
  overflow: hidden;
  ${applyVariants(themeProps.wrapper)};
`

const ItemGroupElement = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 -${p => spacing(p.spacing)} -${p => spacing(p.spacing)};
  & > * {
    margin: 0 ${p => spacing(p.spacing)} ${p => spacing(p.spacing)};
  }
  ${applyVariants(themeProps.group)};
`

ItemGroup.propTypes = {
  /**
   * The component need children inside it.
   */
  children: commonPropTypes.children.isRequired,
  /**
   * The spacing betweeen the children. The prop handle eight types listed above.
   */
  spacing: commonPropTypes.spacing
}

ItemGroup.defaultProps = {
  spacing: "tiny"
}
ItemGroup.themeProps = themeProps
ItemGroup.createVariants = componentCreateFactory(ItemGroup)

export default ItemGroup
