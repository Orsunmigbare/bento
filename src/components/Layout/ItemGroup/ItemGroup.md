The ItemGroup object creates spacing between all its children and wraps the items when there is no room

### Top-aligned

```jsx
<ItemGroup>
  <Button>LÃ¥n</Button>
  <Button>Kreditt</Button>
  <Button>Sparing</Button>
  <Button>Om denne banken</Button>
  <Button>Kontakt</Button>
</ItemGroup>
```
