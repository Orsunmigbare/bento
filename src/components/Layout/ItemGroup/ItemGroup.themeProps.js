import { themePropTypes } from "../../../theming"

const themeProps = {
  wrapper: {
    name: "ITEM_GROUP_WRAPPER",
    description: "",
    type: themePropTypes.style
  },
  group: {
    name: "ITEM_GROUP",
    description: "",
    type: themePropTypes.style
  }
}

export default themeProps
