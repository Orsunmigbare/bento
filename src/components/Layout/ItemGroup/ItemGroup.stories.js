import React from "react"
import { storiesOf } from "@storybook/react"
import docs from "./ItemGroup.md"
import ItemGroup from "./ItemGroup"
import Button from "../../Button/Button"

storiesOf("components|base/Layout/ItemGroup", module)
  .add("No props", () => <ItemGroup />, { info: { text: docs } })
  .add(
    "With  1 button",
    () => {
      return (
        <ItemGroup>
          <Button>Lån</Button>
        </ItemGroup>
      )
    },
    { info: { text: docs } }
  )
  .add(
    "With  some buttons",
    () => {
      return (
        <ItemGroup>
          <Button>Lån</Button>
          <Button>Kreditt</Button>
          <Button>Sparing</Button>
          <Button>Om denne banken</Button>
          <Button>Kontakt</Button>
        </ItemGroup>
      )
    },
    { info: { text: docs } }
  )

  .add(
    "With  many buttons",
    () => {
      return (
        <ItemGroup>
          <Button>Lån</Button>
          <Button>Kreditt</Button>
          <Button>Sparing</Button>
          <Button>Om denne banken</Button>
          <Button>Kontakt</Button>
          <Button>Lån</Button>
          <Button>Kreditt</Button>
          <Button>Sparing</Button>
          <Button>Om denne banken</Button>
          <Button>Kontakt</Button> <Button>Lån</Button>
          <Button>Kreditt</Button>
          <Button>Sparing</Button>
          <Button>Om denne banken</Button>
          <Button>Kontakt</Button>
        </ItemGroup>
      )
    },
    { info: { text: docs } }
  )
