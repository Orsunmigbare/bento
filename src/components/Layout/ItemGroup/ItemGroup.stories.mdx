import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import ItemGroup from "./ItemGroup";

<Meta title="Components|Layout/ItemGroup" component={ItemGroup}/>

# ItemGroup
The ItemGroup object creates spacing between all its children and wraps the items when there is no room.

```javascript
import { ItemGroup } from "@staccx/bento"
```

## Standard ItemGroup
This is the standard ItemGroup. The spacing is default "tiny".

The spacing prop take up to eight types: macro, tiny, small, medium, mediumPlus, large and huge.

<Preview>
  <Story name="Standard">
   <ItemGroup spacing="medium">
        <Button>Loan</Button>
        <Button>Credit</Button>
        <Button>Saving</Button>
        <Button>About this bank</Button>
        <Button>Contact</Button>
   </ItemGroup>
  </Story>
</Preview>


## Props for ItemGroup
<Props of={ItemGroup} />