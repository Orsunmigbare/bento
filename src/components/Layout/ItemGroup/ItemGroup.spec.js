import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import ItemGroup from "./ItemGroup"

const Child = () => <p>Body</p>

describe("Alert", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <ItemGroup>
              <Child />
              <Child />
            </ItemGroup>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Spacing", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <ItemGroup spacing="huge">
              <Child />
              <Child />
            </ItemGroup>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("no children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <ItemGroup />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid spacing", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <ItemGroup spacing={10} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
