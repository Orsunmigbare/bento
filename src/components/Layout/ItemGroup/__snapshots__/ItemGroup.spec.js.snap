// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshots No props 1`] = `
<div
  className="sc-AykKC bIbiJr"
>
  <div
    className="sc-AykKD croRbF"
    spacing="tiny"
  >
    <p>
      Body
    </p>
    <p>
      Body
    </p>
  </div>
</div>
`;

exports[`Alert Snapshots Spacing 1`] = `
<div
  className="sc-AykKC bIbiJr"
>
  <div
    className="sc-AykKD kdSPhR"
    spacing="huge"
  >
    <p>
      Body
    </p>
    <p>
      Body
    </p>
  </div>
</div>
`;

exports[`Alert Snapshots invalid spacing 1`] = `
<div
  className="sc-AykKC bIbiJr"
>
  <div
    className="sc-AykKD fruWvN"
    spacing={10}
  />
</div>
`;

exports[`Alert Snapshots no children 1`] = `
<div
  className="sc-AykKC bIbiJr"
>
  <div
    className="sc-AykKD croRbF"
    spacing="tiny"
  />
</div>
`;
